packages/schema/src/secret/index.ts

```diff
@@ -3,35 +3,39 @@ import { PageRequestSchema, PageResponseSchema } from '@/pagination'
 import { rotateAfterEnum } from '@/enums'
 import { EnvironmentSchema } from '@/environment'
 import { BaseProjectSchema } from '@/project'
-import { WorkspaceSchema } from '@/workspace'
 
 export const SecretSchema = z.object({
-  id: z.string(),
-  name: z.string(),
-  slug: z.string(),
-  createdAt: z.string().datetime(),
-  updatedAt: z.string().datetime(),
-  rotateAt: z.string().datetime().nullable(),
-  note: z.string().nullable(),
-  lastUpdatedById: z.string(),
-  projectId: BaseProjectSchema.shape.id,
-  project: z.object({
-    workspaceId: WorkspaceSchema.shape.id
+  secret: z.object({
+    id: z.string(),
+    name: z.string(),
+    slug: z.string(),
+    createdAt: z.string().datetime(),
+    updatedAt: z.string().datetime(),
+    rotateAt: z.string().datetime().nullable(),
+    note: z.string().nullable(),
+    lastUpdatedById: z.string(),
+    projectId: BaseProjectSchema.shape.id,
+    lastUpdatedBy: z.object({
+      id: z.string(),
+      name: z.string()
+    })
   }),
-  versions: z.array(
+  values: z.array(
     z.object({
-      value: z.string(),
       environment: z.object({
-        id: EnvironmentSchema.shape.id,
-        slug: EnvironmentSchema.shape.slug
-      })
+        id: z.string(),
+        name: z.string(),
+        slug: z.string()
+      }),
+      value: z.string(),
+      version: z.number()
     })
   )
 })
 
 export const CreateSecretRequestSchema = z.object({
   projectSlug: z.string(),
-  name: SecretSchema.shape.name,
+  name: z.string(),
```

Comment:
I'm dropping this comment for just Create Secret, but this wouldd be applicable to all the `name` fields of create and update schemas of all the entities.

We would need to check for this:
- For secrets and variables, charset can only contain `A-Z,a-z.0-9,-,_`
- For others, charset will ALSO contain a whtespace character
- The name must be atleast one character long
- The name must have atleast one character from `A-Z` or `a-z`